src/roman_datamodels/stnode.py

```diff
@@ -110,7 +126,7 @@ class DNode(UserDict):
     def __init__(self, node=None, parent=None, name=None):
         if node is None:
             self.__dict__["_data"] = {}
-        elif isinstance(node, dict):
+        elif isinstance(node, Mapping):
```

Comment:
```suggestion
        elif isinstance(node, dict):
```
I think this can be reverted.